test/mexpr/seq.mc

```diff
@@ -10,43 +10,45 @@ utest [] with [] in
 utest [1,2] with [1,2] in
 utest [[2,3,10],7] with [[2,3,10],7] in
 
-// makeseq n v
-utest makeseq 3 10 with [10,10,10] in
-utest makeseq 8 'a' with ['a','a','a','a','a','a','a','a'] in
-utest makeseq 0 100 with [] in
+// makeSeq n v
+utest makeSeq 3 10 with [10,10,10] in
+utest makeSeq 8 'a' with ['a','a','a','a','a','a','a','a'] in
+utest makeSeq 0 100 with [] in
 
-// concat l1 l2
+// concat s1 s2
 utest concat [1,2,3] [10] with [1,2,3,10] in
 utest concat [1] [3] with [1,3] in
 utest concat [] [3,10] with [3,10] in
 utest concat ['a','b'] [] with ['a','b'] in
 
-// nth lst n
-utest nth [1,3,9] 2 with 9 in
-utest nth [5] 0 with 5 in
-utest nth [5,addi 2 3] 1 with 5 in
+// get seq n
+utest get [1,3,9] 2 with 9 in
+utest get [5] 0 with 5 in
+utest get [5,addi 2 3] 1 with 5 in
+
+// set seq n v
+utest set [1,2,3] 0 4 with [4,2,3] in
+utest set [1] 0 2 with [2] in
+utest set [1,2,3] 2 (addi 1 3) with [1,2,4] in
 
 // cons x xs
 utest cons 1 [8,10] with [1,8,10] in
 utest cons 'a' [] with ['a'] in
 
-// slice lst start length
-utest slice [1,3,5] 0 2 with [1,3] in
-utest slice [3,7,10,20] 1 3 with [7,10,20] in
-utest slice ['a','b'] 1 10 with ['b'] in
-utest slice [1,3] 2 10 with [] in
+// snoc xs x
+utest snoc [1,2] 3 with [1,2,3] in
+utest snoc [] 1 with [1] in
+
+// splitAt seq n
+utest splitAt [1,2,3] 0 with ([],[1,2,3]) in
+utest splitAt [1,2,3] 1 with ([1],[2,3]) in
+utest splitAt [1,2,3] 2 with ([1,2],[3]) in
```

Comment:
I'm surprised that `splitAt [1,2,3] 3` does not equal `([1,2,3, [])` (now it throws an error)! I think this is the behaviour of the finger tree library.

Reply:
`split_at` in batteries has this semantics, but we might want to change this semantics for `splitAt`

Reply:
Change semantics of `splitAt`